src/NHibernate/Engine/Query/CallableParser.cs

```diff
@@ -35,13 +35,13 @@ public static Detail Parse(string sqlString)
 
 			callableDetail.FunctionName = functionMatch.Groups[1].Value;
 
-			callableDetail.HasReturn = sqlString.IndexOf("call") > 0 &&
-										sqlString.IndexOf("?") > 0 &&
-										sqlString.IndexOf("=") > 0 &&
-										sqlString.IndexOf("?") < sqlString.IndexOf("call") &&
-										sqlString.IndexOf("=") < sqlString.IndexOf("call");
+			callableDetail.HasReturn = sqlString.IndexOf("call", StringComparison.Ordinal) > 0 &&
+										sqlString.IndexOf('?') > 0 &&
+										sqlString.IndexOf('=') > 0 &&
+										sqlString.IndexOf('?') < sqlString.IndexOf("call", StringComparison.Ordinal) &&
+										sqlString.IndexOf('=') < sqlString.IndexOf("call", StringComparison.Ordinal);
```

Comment:
We are talking about performance in this PR, right? So `sqlString.IndexOf("call", StringComparison.Ordinal)` needs to be cached in a variable.